### ISSUE TITLE
Fix Windows detection in t.p.lockfile

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5362]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5362|
|Type|defect|
|Created|2011-11-04 20:13:31Z|

Newer Python versions support os.symlink() on recent enough Windows. This patch therefore fixes the Windows detection in twisted.python.lockfile.
(os.symlink() is not reliable under Windows as it requires administrator privileges)

Attachments:

* [winlockfile.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bc7/bc786e95a0e11c478615ec1c93df71f3fe6da439/8aa26f244e68ec311289367ebe93d345b12ae1eb.patch) (627 bytes) - added by antoine on 2011-11-04 20:13:43Z - 
* [winlockfile2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bc7/bc786e95a0e11c478615ec1c93df71f3fe6da439/5254880ea8b8fc0a31fbd024df245d35da182046.patch) (573 bytes) - added by antoine on 2011-11-04 20:29:30Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5362 5362
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1320437611000000 1320437611000000
changetime__1320442798000000 1320442798000000
version__None None
owner__exarkun exarkun
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @pitrou|
|-|-|

Thanks.  We use `twisted.python.runtime` for platform detection wherever possible.  Feel like trying that?

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_3"></a>|@pitrou removed owner|
|-|-|

Indeed. New patch using t.p.runtime.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks!  Looks great.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [33074]) On new Python/Windows combos, avoid trying to make symlinks in twisted.python.lockfile.

Author: antoine
Reviewer: exarkun
Fixes: [#5362](https://github.com/twisted/twisted/issues/5362)

Replace the feature test for `os.symlink` with a platform check.  This makes
the code use the special Windows implementation even when Python exposes an
`os.symlink` API, which newer versions do on recent versions of Windows (but
which requires additional privileges to actually use).
